tools/test/check_syntax.sh

```diff
@@ -14,58 +14,27 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-if [ -z "$SYNTAX_FLAKE_ARGS" ]; then
-    SYNTAX_FLAKE_ARGS="--exclude=\"*.cfg\" bin/* lib/ tools/*.py"
+if [ -z "$SYNTAX_RUFF_ARGS" ]; then
+    SYNTAX_RUFF_ARGS="bin/ lib/ tools/*.py"
 fi
 
-if [ -z "$SYNTAX_PYLINT_ARGS" ]; then
-    SYNTAX_PYLINT_ARGS="lib/rucio/"
-fi
-
-if [ -z "$SYNTAX_PYLINT_BIN_ARGS" ]; then
-    SYNTAX_PYLINT_BIN_ARGS="bin/*"
-fi
-
-
-echo '==============================='
-echo 'Running flake8                 '
-echo '==============================='
-
-flake8 --ignore=E501,E722,W503 $SYNTAX_FLAKE_ARGS
-
-if [ $? -ne 0 ]; then
-    exit 1
-fi
-
-
 echo '==============================='
-echo 'Running pylint                 '
+echo 'Running Ruff                   '
 echo '==============================='
 
-pylint --rcfile=pylintrc --errors-only $SYNTAX_PYLINT_ARGS
+ruff check --extend-ignore=E501,E722 $SYNTAX_RUFF_ARGS
 
 if [ $? -ne 0 ]; then
-    echo "PYLINT on $SYNTAX_PYLINT_ARGS FAILED"
+    echo "Ruff checks failed"
     exit 1
 else
-    echo "PYLINT on $SYNTAX_PYLINT_ARGS PASSED"
+    echo "Ruff checks passed"
 fi
 
-# disable no-name-in-module since bin/rucio clashes with lib/rucio
-PYTHONPATH=lib pylint --rcfile=pylintrc --errors-only --disable no-name-in-module $SYNTAX_PYLINT_BIN_ARGS
-
-if [ $? -ne 0 ]; then
-    echo "PYLINT on $SYNTAX_PYLINT_BIN_ARGS FAILED"
-    exit 1
-else
-    echo "PYLINT on $SYNTAX_PYLINT_BIN_ARGS PASSED"
-fi
-
-
 if [[ "$SYNTAX_REPORT" && "$SYNTAX_REPORT" == "1" ]]; then
     echo '==============================='
-    echo 'Pylint report                  '
+    echo 'Ruff Full Report               '
     echo '==============================='
 
-    pylint --rcfile=pylintrc --reports y --exit-zero $SYNTAX_PYLINT_ARGS
-fi
+    ruff check --output-format=full $SYNTAX_RUFF_ARGS
```

Comment:
Earlier we run ruff with `--extend-ignore=E501,E722`, so this syntax report would be inconsistent with the check done beforehand (extra reason to add them in the main Ruff config instead)

Reply:
Answered in https://github.com/rucio/rucio/pull/7481#discussion_r1999446243